### ISSUE TITLE
Component state & page does not update/rebuild when change in app/component state

### ISSUE THREAD

**Issue by zbabs23** [FlutterFlow/flutterflow-issues#5021]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

The heart toggle in the component "athleteCardSmall" is not asynchronously updated when the associated component state / app state connected to the toggle state value is updated. Only after you refresh the page, it updates correctly. 

### Expected Behavior

When you click on any "athleteCardSmall" component and then toggle the heart icon in the top right, then click the back arrow, the toggle should be updated on the component. Likewise, if you tap the "x" in the saved athletes section on the Dashboard page, it should also asynchronously update the heart toggle on the component below. 

### Steps to Reproduce

1. Go to Dashboard
2. Scroll to the section titled "Find & sign athletes near"
3. Click on the first card
4. A new page should open
5. Click the heart toggle in the top right corner
6. Click on the back arrow in the top left

### Reproducible from Blank

- [ ] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

wrapWithModel(   model: _model.athleteCardSmallModels.getModel(     athleteListItem.id.toString(),     athleteListIndex,   ),   updateCallback: () => safeSetState(() {}),   updateOnChange: true,   child: AthleteCardSmallWidget(     key: Key(       'Key7u9_${athleteListItem.id.toString()}',     ),     athleteAvatar: athleteListItem,     athleteName: athleteListItem,     athleteSchool: athleteListItem,     athleteSport: athleteListItem,     athleteId: athleteListItem,     athleteLocation: athleteListItem,     totalFollowers: athleteListItem,     totalDeals: valueOrDefault<int>(       containerAthleteDealCountsRowList           .where((e) => e.id == athleteListItem.id)           .toList()           .first           .dealCount,       0,     ),     graduationYear: athleteListItem,   ), )

### Visual documentation

<img width="1373" alt="Screenshot 2024-12-02 at 6 06 17 PM" src="https://github.com/user-attachments/assets/4cf010e9-6e65-4b8c-9a13-35ac33b0ded7">

### Environment

```markdown
- FlutterFlow version: v5.0.13
- Platform: Web
- Browser name and version: Chrome 131.0.6778.86
- Operating system and version affected: MacOs 14.3.1 (23D60)
```

**Comment by Alezanello** [comment]
Hello!

To check your project, we’ll need the Bug Report Code generated from the widget tree. You can generate this by right-clicking on any widget in the widget tree and selecting **"Generate Bug Report Code"** from the pop-up menu.

**Comment by zbabs23** [comment]
IT40keflsYt1pbwA+KqJb8FvmmU5JVg7aJ07iu1RaCsofePrOpkAf8/7UFtUcOHjYWw+PmejoDsX/MmOv+GSCsE5NU2tba5Z1btuSzriYGGmRKmYO5S3b3JARJtTIFPG16TQhhJ5PsBYLXgm7V2YPt6vYATqNr7wC20dMeaoXdeK2SrDX1iXc2URm05KZDPz

**Comment by Alezanello** [comment]
Hi there!

Thank you for your update! <br>I still can't be able to see the error,<br>Could you please create a new blank project with just the necessary widgets to replicate the issue? This will help us better understand and address the problem.

Looking forward to your response!

**Comment by zbabs23** [comment]
I have created two new pages to show the issue. Follow these steps:


   1. Go to page /Blank1 and toggle heart icon on one of the widgets under
   section "Find & sign NIL deals near"
   2. A new widget should appear under saved athletes section
   3. Now click on the entire widget where you toggled the heart icon. A
   new page should be displayed with the toggle icon turned ON
   4. Now do the same thing, but instead first click on any of the widgets
   from #1, go to the page and toggle the heart icon on that page
   5. When you return to page /Blank1, the associated widget you clicked in
   #4 should also have the heart icon toggled ON

Please let me know if any questions.


*Zack Babin | *Co-founder & CEO

My calendar <https://calendly.com/zack-babin23/meet-with-zack-babin>
www.modernltv.com
M: 201-835-2155


<http://www.modernwallet.co>

**Comment by Alezanello** [comment]
Hello!

I noticed that in your setup, you are updating the value of the ToggleIcon in the action flow, even though it is already managed by default. Let me clarify how this works:

The value of your ToggleIcon is tied to the variable `isFavorite`. By default, FlutterFlow automatically updates this variable when the icon is toggled. For example, if you don’t add any actions under "On Toggle," clicking the icon will toggle the value of `isFavorite` between `TRUE` and `FALSE`.

However, in your case, it seems you are also setting `isFavorite` within the action flow. This can cause the toggle to switch back and forth unexpectedly because the default behavior and your action are both trying to update the value.

To avoid this issue, you can remove the update 'isFavorite' action from the "On Toggle" flow if the default behavior is sufficient. This will ensure smooth toggling without conflicting updates.<br><br>[https://docs.flutterflow.io/resources/ui/widgets/icons/#on-toggle-action](https://docs.flutterflow.io/resources/ui/widgets/icons/#on-toggle-action)

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/ccc86742-6a42-43a3-a2c9-1c6b2a5f436a/d0f225e5-9bad-4e6c-aae7-9a5a590ec4ca?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS9jY2M4Njc0Mi02YTQyLTQzYTMtYTJjOS0xYzZiMmE1ZjQzNmEvZDBmMjI1ZTUtOWJhZC00ZTZjLWFhZTctOWE1YTU5MGVjNGNhIiwiaWF0IjoxNzM0MTIxNDA3LCJleHAiOjMzMzA0NjgxNDA3fQ.dIqocmOoazKmAhByx1zHKnKyn1NMUDdZJPP5GugU_GI " alt="image.png" width="1196" data-linear-height="176" />

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/c34be7f3-54f6-47db-aae3-e684f5a43c50/97d46790-c025-4d0f-b78c-daa557b18b98?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS9jMzRiZTdmMy01NGY2LTQ3ZGItYWFlMy1lNjg0ZjVhNDNjNTAvOTdkNDY3OTAtYzAyNS00ZDBmLWI3OGMtZGFhNTU3YjE4Yjk4IiwiaWF0IjoxNzM0MTIxNDA3LCJleHAiOjMzMzA0NjgxNDA3fQ.WrCR27hHnEKVm67ZE4fpw-hOZv8PaKEL0wWS-deP3mE " alt="image.png" width="1791" data-linear-height="446" />

**Comment by zbabs23** [comment]
Thanks for highlighting this. I made these changes but the core issue still
remains. The toggle on the widget card does not update unless you refresh
the page. For example when you click on the athlete card widget and open
the new page, then toggle the heart, then go back, you see the toggle is
not updated on the athlete card. Only once you refresh the page.

Please advise

*Zack Babin | *Co-founder & CEO

My calendar <https://calendly.com/zack-babin23/meet-with-zack-babin>
www.modernltv.com
M: 201-835-2155


<http://www.modernwallet.co>

**Comment by Alezanello** [comment]
Perfect! I believe what you need to do is add an **Update App State** action, but without modifying any variables. Then, set the **Rebuild All Pages** option to refresh the page and render the changes effectively.

**Comment by Alezanello** [comment]
Closing this issue since we haven't heard back from you. If you are still facing the problem please submit a new issue. Thank you!
